### ISSUE TITLE
[BUG] .env file at the same level of $COMPOSE_FILE not loaded

### ISSUE THREAD

**Issue by eathtespagheti** [docker/compose#11392]
### Description

Up to version `2.23.3` docker compose would end up loading both the `.env` file of the CWD, and the `.env` file at the same level of the `$COMPOSE_FILE` in use

Since version `2.24.0` only the first `.env` il loaded,  but the other `.env` file at the same level of the `$COMPOSE_FILE` is not sourced, causing those env variables to be empty strings

#### Example on 2.24.0 running `docker compose build`
```
WARN[0000] The "IMAGE_NAME" variable is not set. Defaulting to a blank string. 
WARN[0000] The "IMAGE_TAG" variable is not set. Defaulting to a blank string. 
[+] Building 0.0s (0/0)                                                                                                             docker:default
invalid tag "/:": invalid reference format
```

### Steps To Reproduce

I have a setup with the following tree

```
.
├── .env
└── stack
    ├── development
    │   ├── docker-compose.yaml
    │   ├── .env
    │   └── node.env
    ├── Dockerfile
    └── local
        └── node.env
```

The main `.env` file is automatically loaded if I run `docker compose` in that directory, and this `.env` file set-up the `$COMPOSE_FILE` variable to specify which docker compose file to use

#### `.env`
``` sh
#!/usr/bin/env sh

export COMPOSE_FILE="${COMPOSE_FILE:-stack/development/docker-compose.yaml}"
```

while `stack/development/.env` contains some ENV variables used inside the `stack/development/docker-compose.yaml` file

#### `stack/development/.env`
``` sh
#!/usr/bin/env sh

# Docker
export COMPOSE_PROJECT_NAME=projectname

# Image
export IMAGE_NAME="${IMAGE_NAME:-backend}"
export IMAGE_TAG="${IMAGE_TAG:-latest}"
```

#### `stack/development/docker-compose.yaml`
``` yaml
version: "3.8"

services:

  backend:
    image: $IMAGE_NAME:$IMAGE_TAG
    build:
      context: ../../
      dockerfile: stack/Dockerfile
    restart: unless-stopped
```

### Compose Version

```Text
Docker Compose version 2.24.2
```

### Docker Environment

```Text
Client:
 Version:    24.0.7
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  0.12.1
    Path:     /usr/lib/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  2.24.2
    Path:     /usr/lib/docker/cli-plugins/docker-compose

Server:
 Containers: 7
  Running: 0
  Paused: 0
  Stopped: 7
 Images: 33
 Server Version: 24.0.7
 Storage Driver: btrfs
  Btrfs: 
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 71909c1814c544ac47ab91d2e8b84718e517bb99.m
 runc version: 
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.7.0-arch3-1
 Operating System: Arch Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 14.97GiB
 Name: 8boi
 ID: E3K6:GCEU:3MNV:N4R3:FBUO:HDYW:AW3O:V26O:2LKW:T2LW:EP4C:T5UL
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: abinsula
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by ndeloof** [comment]
This was definitely a bug in v23.x. A single default `.env` file has to be loaded. The fact you have `COMPOSE_FILE` set there and pointing to a distinct folder creates some confusion and a chicken-egg problem to define the expected default location. 

I'm closing this issue as "not planned"

**Comment by eathtespagheti** [comment]
Oks, I will find an alternative then, I just want to point out that this behaviour isn't specific of v23.x, it goes way back at least to v20 when I've started using it, I just never realize this wasn't the expected behaviour

**Comment by eathtespagheti** [comment]
@ndeloof I was checking the [docker compose env file docs](https://docs.docker.com/compose/environment-variables/set-environment-variables/) and it says

> The .env file should be placed at the root of the project directory next to your compose.yaml file.

I think therefore the .env loading up to v23.3 was the correct one, as described by the documentation, making the lack of this functionality a bug in v24, let me know what you think

**Comment by ndeloof** [comment]
@eathtespagheti I wrote this doc :)
this is true as long as project directory is defined by an explicit compose.yaml file being selected. Here we first load a local `.env` file which defines `COMPOSE_FILE` pointing to a distinct folder. Maybe the "right" thing to do would be **not** to load such a local `.env` file as there's no `compose.yaml` file in this folder, so this _should_ not be considered to be project directory

**Comment by rnathuji** [comment]
I've run into a similar issue with version `2.24.1` in a setup that has only a single `.env` file. Oddly, things work when using `-f` but not when setting `COMPOSE_FILE`. Specifically, given:

* A `docker-compose.yaml` at `./mypath/docker-compose.yml`
* A `.env` at `./mypath/.env`

The following results in `WARN` messages due to missing environment variables:

```bash
$ COMPOSE_FILE=./mypath/docker-compose.yml docker compose build
```

The following works fine:

```bash
$ docker compose -f ./mypath/docker-compose.yml build
```

I would expect these two to behave the same given the docs [here](https://docs.docker.com/engine/reference/commandline/compose/) state:

> Setting the COMPOSE_FILE environment variable is equivalent to passing the -f flag

**Comment by eathtespagheti** [comment]
> Maybe the "right" thing to do would be not to load such a local .env file as there's no compose.yaml file in this folder, so this should not be considered to be project directory

@ndeloof yeah this is probably the "right" thing, or at least what I expect to happen reading the docs, but I will add that imho this double loading of .env files is pretty handy, since it allows you to run `docker compose` in a project without having to specify manually the compose file or setting the COMPOSE_FILE variable

@rnathuji yeah I can confirm that on my machine too if I use the -f syntax the corresponding `.env` file is loaded correctly

**Comment by ndeloof** [comment]
@eathtespagheti I agree we should focus on user convenience here, and found a reasonable fix (https://github.com/docker/compose/pull/11405) - which hopefully won't bring additional regressions. Also added a test-case so we get this behavior officially supported.

@rnathuji I can't reproduce with [v2.24.3](https://github.com/docker/compose/releases/tag/v2.24.3)

**Comment by rnathuji** [comment]
:thinking: - @ndeloof It looks like I am still seeing the same behavior with `v2.24.3` downloaded from the releases page.

Just to remove any ambiguity, here is a specific toy example to reproduce:

```
.
└── mypath
    ├── docker-compose.yml
    └── .env
```

Contents of `./mypath/docker-compose.yml`:

```yaml
services:
  postgres:
    image: postgres
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
```

Contents of `./mypath/.env`:

```
POSTGRES_USER=someuser
POSTGRES_PASSWORD=somepassword
POSTGRES_DB=somedb
```

The release binary downloaded and added to my `PATH`:

```bash
$ docker-compose-linux-v2.24.3 version
Docker Compose version v2.24.3
```

The following works:

```bash
$ docker-compose-linux-v2.24.3 -f ./mypath/docker-compose.yml up -d
```

The following emits warnings and the container fails to start:

```bash
$ COMPOSE_FILE=./mypath/docker-compose.yml docker-compose-linux-v2.24.3 up -d
WARN[0000] The "POSTGRES_USER" variable is not set. Defaulting to a blank string.
WARN[0000] The "POSTGRES_PASSWORD" variable is not set. Defaulting to a blank string.
WARN[0000] The "POSTGRES_DB" variable is not set. Defaulting to a blank string.
```

**Comment by ndeloof** [comment]
@rnathuji I can reproduce issue with your example, investigating
